src/genie/libs/parser/asa/show_crypto_ikev2_sa.py

```diff
@@ -0,0 +1,220 @@
+    '''
+
+    cli_command = 'show crypto ikev2 sa'
+
+    def cli(self, output=None):
+        if output is None:
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
+
+        parsed_dict = {}
+        dict_sessions = parsed_dict.setdefault('sessions', {})
+        child_sa_index = 0
+
+        # Session-id:1, Status:UP-ACTIVE, IKE count:1, CHILD count:1
+        p1 = re.compile(
+            r'^Session-id:(?P<session_id>[\d]),\s'
+            r'Status:(?P<status>[A-Za-z\-]{1,}),\s'
+            r'IKE\scount:(?P<ike_count>[\d]),\s'
+            r'CHILD\scount:(?P<child_count>[\d])$'
+        )
+
+        # 3752379 2001:db8:2:1::1/500 2001:db8:2:1::2/500 READY INITIATOR
+        p2 = re.compile(
+            r'^(?P<tunnel_id>\d+)\s+(?P<local>\S+)\s+'
+            r'(?P<remote>\S+)\s+(?P<status>\w+)\s+(?P<role>\w+)$'
+        )
+
+        # Encr: 3DES, Hash: SHA96, DH Grp:2, Auth sign: PSK, Auth verify: PSK
+        # Encr: AES-CBC, keysize: 256, Hash: SHA512, DH Grp:19, Auth sign: PSK, Auth verify: PSK
+        p3 = re.compile(
+            r'^Encr:\s+(?P<encr>\S+),( +keysize:\s+(?P<keysize>\d+),)?'
+            r' +Hash:\s+(?P<hash>\S+),\s+DH\sGrp:(?P<dh_group>\d+),\s+'
+            r'Auth\ssign:\s+(?P<auth_sign>\S+),\s+Auth\sverify:\s+'
+            r'(?P<auth_verify>\S+)$'
+        )
+
+        # Life/Active Time: 43200/53 sec
+        p4 = re.compile(
+            r'^Life\/Active\sTime:\s+'
+            r'(?P<life_time>\d+)\/(?P<active_time>\d+)\s+sec$'
+        )
+
+        # Child sa: local selector  2001:db8:1:1::/0 -
+        # 2001:db8:1:1:ffff:ffff:ffff:ffff/65535
+        p5 = re.compile(
+            r'^Child\ssa:\slocal\sselector\s+(?P<local_selector>.*)$'
+        )
+
+        # remote selector 2001:db8:3:1::/0 -
+        # 2001:db8:3:1:ffff:ffff:ffff:ffff/65535
+        p6 = re.compile(
+            r'^remote\sselector\s+(?P<remote_selector>.*)$'
+        )
+
+        # ESP spi in/out: 0x295bd35a/0x5755a09f
+        p7 = re.compile(
+            r'^ESP\sspi\sin\/out:\s+(?P<esp_in>\S+)\/(?P<esp_out>\S+)$'
+        )
+
+        out = out.splitlines()
+        if not out:
+            return
+
+        for line in out:
+            line = line.strip()
+
+            m = p1.match(line)
+
+            if m:
+                group = m.groupdict()
+                session_id = int(group['session_id'])
+                status = group['status']
+                ike_count = int(group['ike_count'])
+                child_sa_count = int(group['child_count'])
+                dict_session = dict_sessions.setdefault(session_id, {})
+                dict_session.update({'session_id': session_id})
+                dict_session.update({'status': status})
+                dict_session.update({'ike_count': ike_count})
+                dict_session.update({'child_sa_count': child_sa_count})
+                dict_tunnels = dict_session.setdefault('tunnels', {})
+                continue
```

Comment:
Please add a comment to show what line of device output corresponds to this code. It makes debugging and reviewing easier in the long run :)
ex:
```
# Session-id:1, Status:UP-ACTIVE, IKE count:1, CHILD count:1
m = p1.match(line)
if m:
    group = m.groupdict()
    session_id = int(group['session_id'])
    status = group['status']
    ike_count = int(group['ike_count'])
    child_sa_count = int(group['child_count'])
    dict_session = dict_sessions.setdefault(session_id, {})
    dict_session.update({'session_id': session_id})
    dict_session.update({'status': status})
    dict_session.update({'ike_count': ike_count})
    dict_session.update({'child_sa_count': child_sa_count})
    dict_tunnels = dict_session.setdefault('tunnels', {})
    continue
```

Please apply this to all applicable `m = pX.match(line)` lines in all parsers